docs/source/accelerators/accelerator_prepare.rst

```diff
@@ -0,0 +1,169 @@
+.. testsetup:: *
+
+    import torch
+    from pytorch_lightning.trainer.trainer import Trainer
+    from pytorch_lightning.core.lightning import LightningModule
+
+.. _gpu:
+
+########################################
+Hardware agnostic training (preparation)
+########################################
+
+To train on CPU/GPU/TPU without changing your code, we need to build a few good habits :)
+
+----
+
+*****************************
+Delete .cuda() or .to() calls
+*****************************
+
+Delete any calls to .cuda() or .to(device).
+
+.. testcode::
+
+    # before lightning
+    def forward(self, x):
+        x = x.cuda(0)
+        layer_1.cuda(0)
+        x_hat = layer_1(x)
+
+
+    # after lightning
+    def forward(self, x):
+        x_hat = layer_1(x)
+
+----
+
+**********************************************
+Init tensors using type_as and register_buffer
+**********************************************
+When you need to create a new tensor, use `type_as`.
+This will make your code scale to any arbitrary number of GPUs or TPUs with Lightning.
+
+.. testcode::
+
+    # before lightning
+    def forward(self, x):
+        z = torch.Tensor(2, 3)
+        z = z.cuda(0)
+
+
+    # with lightning
+    def forward(self, x):
+        z = torch.Tensor(2, 3)
```

Comment:
wouldn't it be better to use `torch.tensor/torch.empty` with `
dtype=x.dtype
` here? this should be one operation less.

Also should we maybe make use of `self.dtype` and `self.device` here?